tardis/energy_input/gamma_ray_transport.py

```diff
@@ -505,6 +513,36 @@ def decay_chain_energies(
     return decay_energy
 
 
+def fractional_decay_energy(decay_energy):
+    """Function to calculate fractional decay energy
+    Parameters
+    ----------
+    decay_energy : Dict
+        dictionary of decay chain energies for each isotope in each shell
+    Returns
+    -------
+    fractional_decay_energy : Dict
+        dictionary of fractional decay chain energies for each isotope in each shell
+    """
+    fractional_decay_energy = {
+        shell: {
+            parent_isotope: {
+                isotopes: (
+                    decay_energy[shell][parent_isotope][isotopes]
+                    / sum(decay_energy[shell][parent_isotope].values())
+                    if decay_energy[shell][parent_isotope][isotopes] != 0.0
+                    else 0.0
+                )
+                for isotopes in decay_energy[shell][parent_isotope]
+            }
+            for parent_isotope in decay_energy[shell]
+        }
+        for shell in decay_energy
+    }
+
+    return fractional_decay_energy
```

Comment:
that looks complex - would a dataframe be better here?

Reply:
Noted. Thanks!